src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeServerStream.Windows.cs

```diff
@@ -258,17 +258,9 @@ private unsafe IAsyncResult BeginWaitForConnection(AsyncCallback callback, Objec
 
             // Create and store async stream class library specific data in the 
             // async result
-            PipeAsyncResult asyncResult = new PipeAsyncResult();
-            asyncResult._threadPoolBinding = _threadPoolBinding;
-            asyncResult._userCallback = callback;
-            asyncResult._userStateObject = state;
-
+            PipeAsyncResult asyncResult = new PipeAsyncResult(_threadPoolBinding, callback, state);
```

Comment:
As evidenced by the code coverage report, we have very little test coverage for this type and code paths using it.  I don't want to take such a change without such tests.  Can you write some to go along with this?  It's of course more than just about ensuring lines are executed, but about making sure everything behaves the way we expect.  I'd prefer to see a commit that adds a bunch of such tests for various conditions, verifying that all such tests pass, and then subsequently a commit like this one that attempts to improve the implementation.